src/main/java/com/enonic/xp/app/welcome/WelcomePageScriptBean.java

```diff
@@ -202,19 +215,101 @@ public Object getSites()
     public Object getProjects()
     {
         List<ProjectJson> projects = new ArrayList<>();
-        projects.addAll( createAdminContext( ContentConstants.CONTENT_REPO_ID ).
-            callWith( () -> projectServiceSupplier.get().list() ).
-            stream().
-            sorted( Comparator.comparing( project -> project.getName().getRepoId().toString() ) ).
-            map( project -> ProjectJson.create().
-                project( project ).
-                iconAsBase64( getProjectIconAsBase64( project ) ).
-                build() ).
-            collect( Collectors.toList() ) );
+        projects.addAll(
+            createAdminContext( ContentConstants.CONTENT_REPO_ID ).callWith( () -> projectServiceSupplier.get().list() ).stream().sorted(
+                Comparator.comparing( project -> project.getName().getRepoId().toString() ) ).map(
+                project -> ProjectJson.create().project( project ).iconAsBase64( getProjectIconAsBase64( project ) ).build() ).collect(
+                Collectors.toList() ) );
 
         return new ProjectsMapper( projects );
     }
 
+    public Object getTemplateApplications()
+    {
+        List<TemplateApplicationJson> templateApps = new ArrayList<>();
+
+        Path filePath = HomeDir.get().toPath().resolve( "config" ).resolve( ".template" );
+        if ( Files.exists( filePath ) )
+        {
+            try
+            {
+                templateApps.addAll( mustParseJsonFile( filePath.toString() ) );
+            }
+            catch ( Exception e )
+            {
+                LOG.error( "Could not parse template file", e );
+            }
+        }
+
+        return new TemplateApplicationsMapper( templateApps );
+    }
+
+    public boolean deleteTemplateFile()
+    {
+        Path filePath = HomeDir.get().toPath().resolve( "config" ).resolve( ".template" );
+        if ( Files.exists( filePath ) )
+        {
+            try
+            {
+                Files.delete( filePath );
+                return true;
+            }
+            catch ( IOException e )
+            {
+                throw new UncheckedIOException( "Could not delete template applications file at " + filePath, e );
```

Comment:
if file is read-only, this will throw an exception. Since it is not a critical procedure it is better to log a warning about it, not crash and burn.